### PR TITLE
Create Draft Permit

### PR DESCRIPTION
**Pull request by mmumby**
- Ability to create draft permit amendments through two paths: New Permits and Amendments
- If it is a " New Permit" a permit number is generated - logic will be updated in a following ticket -if its of type "Mineral" or "Coal" the user must say if its an exploration permit or not  before creating the draft permit, if it is a different type, the permit is created without this gate.
Screen Shot 2020-07-23 at 9 28 35 AM
- If the draft is an "Amendment", the user must select which permit it belongs to given a dropdown list of all possible permits associated with the mine.
- Ability to edit the amendments once created
- All fields are disabled and information is pulled from the NoW with the exception of region and conditions - if the user wants to update other fields they must edit through the NoW
Screen Shot 2020-07-23 at 9 13 24 AM
Screen Shot 2020-07-23 at 9 13 41 AM
Screen Shot 2020-07-23 at 9 15 58 AM
Screen Shot 2020-07-23 at 9 16 14 AM
Screen Shot 2020-07-23 at 9 18 03 AM

- removed working permit/working permit amendment generate options from the admin dropdown.
Screen Shot 2020-07-23 at 9 18 24 AM

### CODE REVIEW

File: services/core-api/app/api/parties/party_appt/resources/mine_party_appt_resource.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def get(self, mine_party_appt_guid=None):
                 raise NotFound('Mine Party Appointment not found')
             result = mpa.json(relationships=relationships)
         else:
+            current_app.logger.info(request.args)
```
Comment (Jsyro) [inline services/core-api/app/api/parties/party_appt/resources/mine_party_appt_resource.py:60]: This came from my WIP, please remove.

File: services/core-api/app/api/mines/permits/permit/models/permit_status_code.py [hunk 2]
```diff
@@ -19,7 +21,7 @@ def find_by_permit_status_code(cls, _id):
 
     @classmethod
     def get_all(cls):
-        return cls.query.all()
+        return cls.query.filter_by(active_ind=True).all()
         #TODO put active ind on this table
```
[the remark this answers is not in the document]
Comment (mmumby) [reply services/core-api/app/api/mines/permits/permit/models/permit_status_code.py:25]: removed

File: services/core-api/app/api/mines/permits/permit/models/permit.py [hunk 3]
```diff
@@ -126,6 +130,13 @@ def find_by_permit_guid_or_no(cls, _permit_guid_or_no):
             result = cls.find_by_permit_no(_permit_guid_or_no)
         return result
 
+    @classmethod
+    def find_by_now_application_guid(cls, _now_application_guid):
+        permit_amendment = PermitAmendment.find_by_now_application_guid(_now_application_guid)
+        permit = cls.find_by_permit_id(permit_amendment.permit_id)
```
[the remark this answers is not in the document]
Comment (mmumby) [reply services/core-api/app/api/mines/permits/permit/models/permit.py:136]: I'm not sure what line I'm removing here.

@nayr974 ?

File: services/core-api/app/api/parties/party_appt/resources/mine_party_appt_resource.py (hunk 1 above)
Comment (mmumby) [reply services/core-api/app/api/parties/party_appt/resources/mine_party_appt_resource.py:60]: removed

File: services/core-api/app/api/mines/mine/models/mine.py [hunk 4]
```diff
@@ -146,7 +146,10 @@ def mine_permit(self):
 
     @hybrid_property
     def mine_permit_numbers(self):
-        p_numbers = [mpi.permit_no for mpi in self._permit_identities]
+        p_numbers = []
+        for p in self._permit_identities:
+            if p.deleted_ind == False and p.permit_status_code != 'D':
```
[the remark this answers is not in the document]
Comment (mmumby) [reply services/core-api/app/api/mines/mine/models/mine.py]: updated!

File: services/core-api/app/api/mines/permits/permit/models/permit.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jsyro) [reply services/core-api/app/api/mines/permits/permit/models/permit.py:136]: permit_amendment.permit will give you the same permit as line 136 but doesn't lets the ORM figure out how to get it.

[the remark this answers is not in the document]
Comment (mmumby) [reply services/core-api/app/api/mines/permits/permit/models/permit.py:136]: It seems odd to me that the `permit` object is nested under `permit_amendment`

```
    @classmethod
    def find_by_now_application_guid(cls, _now_application_guid):
        permit_amendment = PermitAmendment.find_by_now_application_guid(_now_application_guid)
        permit = permit_amendment.permit
        permit._context_mine = permit_amendment.mine
        return permit
```

[the remark this answers is not in the document]
Comment (mmumby) [reply services/core-api/app/api/mines/permits/permit/models/permit.py:136]: seems to work fine.

File: services/core-web/common/selectors/permitSelectors.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable */
+import { createSelector } from "reselect";
+import { getNoticeOfWork } from "@common/selectors/noticeOfWorkSelectors";
+import * as permitReducer from "../reducers/permitReducer";
+
+export const { getPermits, getDraftPermits } = permitReducer;
+
+export const getDraftPermitForNOW = createSelector(
```
Comment (nayr974) [inline services/core-web/common/selectors/permitSelectors.js:8]: great function

File: services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js [hunk 6]
```diff
@@ -1,29 +1,126 @@
...
+  componentDidMount() {
+    const permittee = this.props.noticeOfWork.contacts.filter(
+      (contact) => contact.mine_party_appt_type_code_description === "Permittee"
+    )[0];
+    this.setState({ permittee });
+    this.props.fetchPermits(this.props.noticeOfWork.mine_guid);
+    this.handleDraftPermit();
+  }
+
+  handleDraftPermit = () => {
```
Comment (nayr974) [inline services/core-web/src/components/noticeOfWork/applications/permitGeneration/NOWPermitGeneration.js:77]: If we continue working on this file we should probably break some of these out into other files as stateless components, as it's getting pretty big.

**Pull request merged**